main.go

```diff
@@ -161,12 +161,21 @@ func main() {
 	}
 
 	maxInterval := *interval
-
+	var serverProtocol string
+	var httpsCertFile string
+	var httpsKeyFile string
 	sc.Lock()
 	for _, targetGroup := range sc.C.Targets {
 		if targetGroup.Interval > maxInterval {
 			maxInterval = targetGroup.Interval
 		}
+		serverProtocol = targetGroup.ServerProtocol
+		if serverProtocol != "http" && serverProtocol != "https" {
+			level.Error(logger).Log("msg", "Invalid server protocol", serverProtocol)
+			return
+		}
+		httpsCertFile = targetGroup.HttpsCertFile
+		httpsKeyFile = targetGroup.HttpsKeyFile
```

Comment:
There is only one http server for the whole tool, why is this in the targets list?

This also means you can't configure TLS when specifying targets on the command args.